packages/davinci-client/src/types.ts

```diff
@@ -1,3 +1,5 @@
+import 'immer'; // Side-effect needed only for getting types in workspace
```

Comment:
This feels like a smell. We would have to mark this as a `side effect` in the package.json.

If this is purely a type related need there should be better ways of doing it.

A) I think all types should be default included unless a `types` field is included. So i think if we have a `types` field in the package.json maybe we should remove it, or we should add this type to the field.

Is the expected behavior just that the `immer` types are scoped in this module?

Reply:
I guess i'm confused as what the purpose is, Why are we trying to populate the module scope with the immer types?

Reply:
I remember having a conversation with you about this a while ago. I think it's from instances of the `createSlice`. Without getting the `immer` types into the scope, the variable assigned a `createSlice` has to have an explicit, generic (not `<T>` type generic, but just generic as in plain) type on it, which then removes the implicit type, which contains the specific attributes of that particular slice. I hope that makes sense.

Maybe you'll understand it better than me, so I'd encourage you to pull this yourself and remove the immer stuff. Checkout the slices to see the error that should be present in your IDE.